gax-grpc/src/test/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProviderTest.java

```diff
@@ -157,6 +157,45 @@ public void testWithPoolSize() throws IOException {
     }
   }
 
+  @Test
+  public void testToBuilder() {
+    Duration keepaliveTime = Duration.ofSeconds(1);
+    Duration keepaliveTimeout = Duration.ofSeconds(2);
+    ApiFunction<ManagedChannelBuilder, ManagedChannelBuilder> channelConfigurator =
+        new ApiFunction<ManagedChannelBuilder, ManagedChannelBuilder>() {
+          @Override
+          public ManagedChannelBuilder apply(ManagedChannelBuilder input) {
+            throw new UnsupportedOperationException();
+          }
+        };
+    Map<String, ?> directPathServiceConfig = ImmutableMap.of("loadbalancingConfig", "grpclb");
+
+    InstantiatingGrpcChannelProvider provider =
```

Comment:
does this class have a good equals method? that could make for a more robust test.

Reply:
`InstantiatingGrpcChannelProvider` has complex logic such as relation between PoolSize and ProcessorCount and ChannelsPerCpu, and attributes depending on system properties and env. It might not make much sense to simply implement `equals()` by comparing its fields, just for testing purpose.